tensorflow_addons/layers/embedding_bag.py

```diff
@@ -0,0 +1,165 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+import tensorflow as tf
+from typeguard import typechecked
+
+from tensorflow_addons.utils.types import Constraint, Initializer, Regularizer
+from tensorflow_addons.utils.resource_loader import LazySO
+
+_embedding_bag_so = LazySO("custom_ops/layers/_embedding_bag_ops.so")
+
+
+def _embedding_bag(
+    indices,
+    params,
+    weights=None,
+    combiner="sum",
+    name=None,
+):
+    """EmbeddingBag computation.
+
+    See [PyTorch op](https://pytorch.org/docs/stable/generated/torch.nn.EmbeddingBag.html).
+
+    Equivalent to tf.gather() followed by tf.reduce_{sum,mean}() across the last dimension, with optional
+    weights. Fusing these into a single op has massive benefits for execution speed and particularly
+    memory usage, as the intermediate output of the gather never needs to be materialized.
+
+    Args:
+      indices: An int32 or int64 `Tensor` of the indices to gather from
+          `params`. Must be at least 2-dimensional, as the last dimension
+          will be summed out. Maximum value must be less than params.shape[0].
+      params: A float32 `Tensor` from which to gather params. Must be rank 2.
+      weights: A float32 `Tensor` of weights which will be applied to each of
+          the gathered embedding vectors before the sum step.
+      name: A name for the operation (optional).
+
+    Returns:
+      A `Tensor` of the format specified by `data_format`.
+    """
+    if weights is None:
+        weights = tf.ones_like(indices, dtype=params.dtype)
+    elif combiner != "sum":
+        raise RuntimeError(
+            "Combiner mode must be 'sum' when weights are supplied to EmbeddingBag!"
+        )
+
+    return _embedding_bag_so.ops.addons_embedding_bag(
+        indices, params, weights, combiner=combiner.upper(), name=name
+    )
+
+
+@tf.RegisterGradient("Addons>EmbeddingBag")
+def _embedding_bag_grad(op, grads):
+    indices, params, weights = op.inputs[:3]
+    combiner = op.get_attr("combiner")
+    value_grads, weight_grads = _embedding_bag_so.ops.addons_embedding_bag_grad(
+        indices, params, weights, grads, combiner=combiner
+    )
+    # Because value grads are sparse, returning IndexedSlices can be faster for optimizer.
+    unique_indices = tf.unique(tf.reshape(indices, (-1,)))[0]
+    sorted_unique_indices = tf.sort(unique_indices)
+    return [
+        None,
+        tf.IndexedSlices(
```

Comment:
Maybe just return value_grads is a better choice, according to https://github.com/foxik/tensorflow/blob/b0e08f68d84476b941e8a821d653ab08129c92ce/tensorflow/python/eager/function.py#L1241.

Reply:
Good idea, I made the change!